test/lib/ufe/testDeleteCmd.py

```diff
@@ -203,6 +206,113 @@ def testDelete(self):
+        propsItem = ufe.Hierarchy.createItem(propsPath)
+
+        sphereShapeName = str(sphereShapeItem.path().back())
+        ball35Name = str(ball35Item.path().back())
+
+        ufe.GlobalSelection.get().append(sphereShapeItem)
+        ufe.GlobalSelection.get().append(ball35Item)
+
+        # Before delete, each item is a child of its parent.
+        sphereHierarchy = ufe.Hierarchy.hierarchy(sphereItem)
+        propsHierarchy = ufe.Hierarchy.hierarchy(propsItem)
+
+        sphereChildren = sphereHierarchy.children()
+        propsChildren = propsHierarchy.children()
+
+        sphereChildrenNames = childrenNames(sphereChildren)
+        propsChildrenNames = childrenNames(propsChildren)
+
+        self.assertIn(sphereShapeItem, sphereChildren)
+        self.assertIn(ball35Item, propsChildren)
+        self.assertIn(sphereShapeName, sphereChildrenNames)
+        self.assertIn(ball35Name, propsChildrenNames)
+
+        ufeObs.reset()
+        cmds.delete()
+
+        # We only deleted the maya object.
+        self.assertEqual(ufeObs.nbDeleteNotif(), 1)
+        self.assertEqual(ufeObs.nbAddNotif(), 0)
+
+        sphereChildren = sphereHierarchy.children()
+        propsChildren = propsHierarchy.children()
+
+        sphereChildrenNames = childrenNames(sphereChildren)
+        propsChildrenNames = childrenNames(propsChildren)
+
+        self.assertNotIn(sphereShapeName, sphereChildrenNames)
+        # ball_35 cannot be deleted check its still in list
+        self.assertIn(ball35Name, propsChildrenNames)
+
+        cmds.undo()
+
+        # After the undo we added one items back.
+        self.assertEqual(ufeObs.nbDeleteNotif(), 1)
+        self.assertEqual(ufeObs.nbAddNotif(), 1)
+
+        sphereChildren = sphereHierarchy.children()
+        propsChildren = propsHierarchy.children()
+
+        sphereChildrenNames = childrenNames(sphereChildren)
+        propsChildrenNames = childrenNames(propsChildren)
+
+        self.assertIn(sphereShapeItem, sphereChildren)
+        self.assertIn(ball35Item, propsChildren)
+        self.assertIn(sphereShapeName, sphereChildrenNames)
+        self.assertIn(ball35Name, propsChildrenNames)
+        
+        cmds.redo()
+
+        # After the redo we again deleted one item.
+        self.assertEqual(ufeObs.nbDeleteNotif(), 2)
+        self.assertEqual(ufeObs.nbAddNotif(), 1)
+
+        sphereChildren = sphereHierarchy.children()
+        propsChildren = propsHierarchy.children()
+
+        sphereChildrenNames = childrenNames(sphereChildren)
+        propsChildrenNames = childrenNames(propsChildren)
+
+        self.assertNotIn(sphereShapeName, sphereChildrenNames)
+        self.assertIn(ball35Name, propsChildrenNames)
+
+        # undo to restore state to original.
+        cmds.undo()
+
+        # After the undo we again added the sphere item back.
+        self.assertEqual(ufeObs.nbDeleteNotif(), 2)
+        self.assertEqual(ufeObs.nbAddNotif(), 2)
```

Comment:
This test feels awkward, because in essence it's testing deleting a Maya object, which Maya already tests.  If the point of this test is to demonstrate we can't delete a referenced object, then we can test that, and forget about undo and redo.

Reply:
good point, removed undo and redo test